packages/app-page-builder/src/admin/plugins/pageDetails/header/Header.tsx

```diff
@@ -2,46 +2,100 @@ import React from "react";
+import { ReactComponent as LinkIcon } from "@material-design-icons/svg/round/link.svg";
 import { PbPageData } from "~/types";
 
-const HeaderTitle = styled("div")({
-    display: "flex",
-    alignItems: "center",
-    justifyContent: "space-between",
-    borderBottom: "1px solid var(--mdc-theme-on-background)",
-    color: "var(--mdc-theme-text-primary-on-background)",
-    background: "var(--mdc-theme-surface)",
-    paddingTop: 10,
-    paddingBottom: 9,
-    paddingLeft: 24,
-    paddingRight: 24
-});
-
-const PageTitle = styled("div")({
-    whiteSpace: "nowrap",
-    overflow: "hidden",
-    textOverflow: "ellipsis"
-});
-
-const HeaderActions = styled("div")({
-    justifyContent: "flex-end",
-    marginRight: "-15px",
-    marginLeft: "10px",
-    display: "flex",
-    alignItems: "center"
-});
+const HeaderTitle = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    border-bottom: 1px solid var(--mdc-theme-on-background);
+    color: var(--mdc-theme-text-primary-on-background);
+    background: var(--mdc-theme-surface);
+    padding-top: 10px;
+    padding-bottom: 9px;
+    padding-left: 24px;
+    padding-right: 24px;
+`;
+
+const PageInfo = styled.div`
+    display: flex;
+    flex-direction: column;
+    min-width: 0;
+`;
+
+const PageTitle = styled.div`
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+
+    span {
+        font-weight: bold;
+    }
+`;
+
+const PageLink = styled.div`
+    display: flex;
+    align-items: center;
+    height: 24px;
+
+    span {
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        max-width: 300px;
+    }
+
+    svg {
+        display: none;
+        flex-shrink: 0;
+        width: 20px;
+        height: 20px;
+        margin-left: 8px;
+        cursor: pointer;
+    }
+
+    :hover svg {
+        display: block;
+    }
+`;
+
+const HeaderActions = styled.div`
+    justify-content: flex-end;
+    margin-right: -15px;
+    margin-left: 10px;
+    display: flex;
+    align-items: center;
+`;
 
 interface HeaderProps {
     page: PbPageData;
 }
 const Header: React.FC<HeaderProps> = props => {
     const { page } = props;
+    const { getPageUrl, getWebsiteUrl } = usePageBuilderSettings();
+    const [isSiteRunning] = useSiteStatus(getWebsiteUrl());
+
+    // We must prevent opening in new tab - Cypress doesn't work with new tabs.
+    const target = "Cypress" in window ? "_self" : "_blank";
+    const url = getPageUrl(page);
+
     return (
         <React.Fragment>
             <HeaderTitle>
-                <PageTitle>
-                    <Typography use="headline5">{page.title}</Typography>
-                </PageTitle>
+                <PageInfo>
+                    <PageTitle>
+                        <Typography use="headline6">{page.title}</Typography>
+                    </PageTitle>
+                    {isSiteRunning && (
```

Comment:
Can we rather have the link always shown, but when clicking on it, if the site is not running, display that modal that we show in other places? Like when we click on the View page in dropdown:
![image](https://user-images.githubusercontent.com/5121148/226864557-5e0fb62c-4b29-44dd-a980-a8a088de2f03.png)